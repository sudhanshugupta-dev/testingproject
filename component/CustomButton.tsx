import { PropsWithChildren } from 'react';
import { Pressable, Text, StyleSheet, TouchableOpacity } from 'react-native';

export const CustomButton = ({
  children,
  onPress,
}: PropsWithChildren<{ onPress: () => void }>) => {
  return (
    <TouchableOpacity style={styles.button} onPress={onPress}>
      <Text style={styles.text}>{children}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#007AFF',
    padding: 10,
    height: '60%',
    width: '60%',
    borderRadius: 15,
  },
  text: {
    color: 'white',
    textAlign: 'center',
  },
});
