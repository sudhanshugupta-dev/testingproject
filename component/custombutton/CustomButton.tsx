import { PropsWithChildren } from 'react';
import { Pressable, Text, StyleSheet, TouchableOpacity } from 'react-native';
import styles from './Styles';

export const CustomButton = ({
  children,
  onPress,
}: PropsWithChildren<{ onPress: () => void }>) => {
  return (
    <TouchableOpacity style={styles.button} onPress={onPress}>
      <Text style={styles.text}>{children}</Text>
    </TouchableOpacity>
  );
};

