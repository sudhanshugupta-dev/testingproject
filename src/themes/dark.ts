export default {
  background: '#0B1020',
  text: '#E5E7EB',
  primary: '#6366F1',
  accent: '#34D399',
  card: '#111827',
  danger: '#F87171'
};
