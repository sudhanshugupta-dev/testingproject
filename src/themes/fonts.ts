export const fonts = {
  weight: { regular: '400', medium: '500', bold: '700' },
  sizes: { small: 12, medium: 16, large: 20, xlarge: 24 }
};
