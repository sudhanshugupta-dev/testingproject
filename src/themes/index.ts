// Main themes export file - makes themes accessible from anywhere
export { useAppTheme, getThemeColors, getTheme } from './useTheme';
export { default as lightColors } from './light';
export { default as darkColors } from './dark';
export { fonts } from './fonts';

// Types
export type ThemeColors = typeof import('./light').default;
export type ThemeFonts = typeof import('./fonts').fonts;

export interface Theme {
  colors: ThemeColors;
  fonts: ThemeFonts;
  mode: 'light' | 'dark';
}
