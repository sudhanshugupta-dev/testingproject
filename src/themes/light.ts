export default {
  background: '#FFFFFF',
  surface: '#FAFBFC',
  text: '#1F2937',
  textSecondary: '#6B7280',
  primary: '#6366F1',
  primaryLight: '#A5B4FC',
  secondary: '#10B981',
  accent: '#F59E0B',
  card: '#F9FAFB',
  cardBorder: '#E5E7EB',
  danger: '#EF4444',
  warning: '#F59E0B',
  success: '#10B981',
  info: '#3B82F6',
  shadow: '#00000020',
  border: '#E5E7EB',
  placeholder: '#9CA3AF',
  disabled: '#D1D5DB',
  overlay: '#00000050'
};
