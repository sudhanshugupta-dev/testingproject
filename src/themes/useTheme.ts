import { useSelector } from 'react-redux';
import { RootState } from '../redux/store';
import light from './light';
import dark from './dark';
import { fonts } from './fonts';

export const useAppTheme = () => {
  const mode = useSelector((s: RootState) => s.theme.mode);
  const colors = mode === 'dark' ? dark : light;
  return { mode, colors, fonts } as const;
}; 