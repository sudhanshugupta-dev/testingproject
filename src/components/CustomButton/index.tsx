import React from 'react';
import { Pressable, Text, ViewStyle, StyleSheet } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import { useAppTheme } from '../../themes/useTheme';

type Props = { title: string; onPress?: () => void; style?: ViewStyle; disabled?: boolean; testID?:String };

const CustomButton = ({ title, onPress, style, disabled }: Props) => {
  const { fonts } = useAppTheme();
  return (
    <Pressable disabled={disabled} testID={'custom-button'} onPress={onPress} style={({ pressed }) => [{ opacity: pressed ? 0.9 : 1 }, style]}>
      <LinearGradient colors={['#6366F1', '#22C55E']} start={{ x: 0, y: 0 }} end={{ x: 1, y: 0 }} style={styles.btn}>
        <Text style={[styles.text, { fontWeight: fonts.weight.bold as any, fontSize: fonts.sizes.large }]}>{title}</Text>
      </LinearGradient>
    </Pressable>
  );
};

const styles = StyleSheet.create({ btn: { paddingVertical: 12, borderRadius: 12, alignItems: 'center' }, text: { color: '#fff' } });

export default CustomButton;
