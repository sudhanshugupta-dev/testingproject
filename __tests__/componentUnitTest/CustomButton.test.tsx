import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import CustomButton from '../../src/components/CustomButton';
import { useAppTheme } from '../../src/themes/useTheme';

// Mock the theme hook
jest.mock('../../src/themes/useTheme', () => ({
  useAppTheme: jest.fn(),
}));

// Mock LinearGradient
jest.mock('react-native-linear-gradient', () => 'LinearGradient');

// Mock Redux store
const mockStore = configureStore({
  reducer: {
    theme: (state = { mode: 'light' }) => state,
  },
});

const mockTheme = {
  colors: {
    primary: '#007AFF',
    primaryLight: '#4DA3FF',
    secondary: '#5856D6',
    accent: '#FF2D92',
    danger: '#FF3B30',
  },
  fonts: {
    sizes: {
      small: 12,
      medium: 14,
      large: 16,
    },
    weight: {
      normal: '400',
      medium: '500',
      bold: '700',
    },
  },
};

describe('CustomButton Component', () => {
  const mockOnPress = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
    (useAppTheme as jest.Mock).mockReturnValue(mockTheme);
  });

  it('renders with correct title', () => {
    const { getByText } = render(
      <Provider store={mockStore}>
        <CustomButton title="Test Button" onPress={mockOnPress} />
      </Provider>
    );

    expect(getByText('Test Button')).toBeTruthy();
  });

  it('calls onPress when pressed', () => {
    const { getByTestId } = render(
      <Provider store={mockStore}>
        <CustomButton title="Test Button" onPress={mockOnPress} />
      </Provider>
    );

    fireEvent.press(getByTestId('custom-button'));
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it('does not call onPress when disabled', () => {
    const { getByTestId } = render(
      <Provider store={mockStore}>
        <CustomButton title="Test Button" onPress={mockOnPress} disabled={true} />
      </Provider>
    );

    fireEvent.press(getByTestId('custom-button'));
    expect(mockOnPress).not.toHaveBeenCalled();
  });

  it('does not call onPress when loading', () => {
    const { getByTestId } = render(
      <Provider store={mockStore}>
        <CustomButton title="Test Button" onPress={mockOnPress} loading={true} />
      </Provider>
    );

    fireEvent.press(getByTestId('custom-button'));
    expect(mockOnPress).not.toHaveBeenCalled();
  });

  it('shows loading indicator when loading is true', () => {
    const { getByTestId } = render(
      <Provider store={mockStore}>
        <CustomButton title="Test Button" onPress={mockOnPress} loading={true} />
      </Provider>
    );

    expect(getByTestId('loading-indicator')).toBeTruthy();
  });

  it('applies custom style', () => {
    const customStyle = { marginTop: 20 };
    const { getByTestId, UNSAFE_root } = render(
      <Provider store={mockStore}>
        <CustomButton title="Test Button" onPress={mockOnPress} style={customStyle} />
      </Provider>
    );

    // Find the outer View container that receives the style prop
    const container = UNSAFE_root.findByType('View');
    expect(container.props.style).toEqual(
      expect.arrayContaining([
        expect.objectContaining(customStyle)
      ])
    );
  });

  it('renders with primary variant by default', () => {
    const { getByTestId } = render(
      <Provider store={mockStore}>
        <CustomButton title="Test Button" onPress={mockOnPress} />
      </Provider>
    );

    const button = getByTestId('custom-button');
    expect(button).toBeTruthy();
  });

  it('renders with secondary variant', () => {
    const { getByTestId } = render(
      <Provider store={mockStore}>
        <CustomButton title="Test Button" onPress={mockOnPress} variant="secondary" />
      </Provider>
    );

    const button = getByTestId('custom-button');
    expect(button).toBeTruthy();
  });

  it('renders with danger variant', () => {
    const { getByTestId } = render(
      <Provider store={mockStore}>
        <CustomButton title="Test Button" onPress={mockOnPress} variant="danger" />
      </Provider>
    );

    const button = getByTestId('custom-button');
    expect(button).toBeTruthy();
  });

  it('applies disabled styles when disabled', () => {
    const { getByTestId } = render(
      <Provider store={mockStore}>
        <CustomButton title="Test Button" onPress={mockOnPress} disabled={true} />
      </Provider>
    );

    // Check that the button is rendered and the disabled functionality works
    // (which is already tested in the "does not call onPress when disabled" test)
    const button = getByTestId('custom-button');
    expect(button).toBeTruthy();
    
    // Test that pressing the button doesn't call onPress (disabled behavior)
    fireEvent.press(button);
    expect(mockOnPress).not.toHaveBeenCalled();
  });

  it('handles press without onPress prop', () => {
    const { getByTestId } = render(
      <Provider store={mockStore}>
        <CustomButton title="Test Button" />
      </Provider>
    );

    const button = getByTestId('custom-button');
    expect(() => fireEvent.press(button)).not.toThrow();
  });
}); 