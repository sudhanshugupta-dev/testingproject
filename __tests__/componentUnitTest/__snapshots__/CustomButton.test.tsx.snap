// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "opacity": 1,
      },
      undefined,
    ]
  }
  testID="custom-button"
>
  <BVLinearGradient
    colors={
      [
        4284704497,
        4280468830,
      ]
    }
    endPoint={
      {
        "x": 1,
        "y": 0,
      }
    }
    locations={null}
    startPoint={
      {
        "x": 0,
        "y": 0,
      }
    }
    style={
      {
        "alignItems": "center",
        "borderRadius": 12,
        "paddingVertical": 12,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#fff",
          },
          {
            "fontSize": 20,
            "fontWeight": "700",
          },
        ]
      }
    >
      Snapshot
    </Text>
  </BVLinearGradient>
</View>
`;
